recipes/libcurl/all/conanfile.py

```diff
@@ -213,13 +206,28 @@ def build_requirements(self):
         else:
             self.tool_requires("libtool/2.4.6")
             self.tool_requires("pkgconf/1.7.4")
-            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+            if self._settings_build.os == "Windows" and not get_env("CONAN_BASH_PATH") and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
                 self.tool_requires("msys2/cci.latest")
 
+    def layout(self):
+        if self._is_using_cmake_build:
```

Comment:
You can not conditionally set the layout.

This sets up important things on the package info... which AFAIK could explain why the test package does not work

Reply:
Use `basic_layout` if not self._is_using_cmake_build

Reply:
New advice 😱

Reply:
Important: define a consistent src folder if there are conditional layouts.

Reply:
I asked about conditional layout here: https://github.com/conan-io/conan/issues/11876